### PR TITLE
Add user arg for docker driver

### PR DESCRIPTION
**Pull request by ssbarnea**
Allow 'user' to be specified when providing arguments to the docker driver.

Originally raised at https://github.com/ansible/molecule/pull/1526 but it got outdated and I was not able to rebase it there.

#### PR Type

- Feature Pull Request

### CODE REVIEW

**Comment by ssbarnea** [comment]
@logan2211 Please check ^^

**Comment by rockandska** [comment]
This PR is one among another concerning missing module arguments that i solved by only rewrite a create/destroy who could be found here https://github.com/ansible/molecule/issues/1880#issuecomment-477522088 if there's some people interesting to test it and comment

File: molecule/driver/docker.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ class Docker(base.Base):
                 username: $USERNAME
                 password: $PASSWORD
                 email: user@example.com
+			user: root
```
Comment (logan2211) [inline molecule/driver/docker.py]: Possible indentation issue?

**Comment by ssbarnea** [comment]
Please review this.

**Comment by ssbarnea** [comment]
> Please add a test for the schema validation.

Schema validation is going to be dropped as we cannot keep-up with docker module changes. Instead we will pass all arguments to docker modules, and making molecule more future proof.

File: test/unit/model/v2/test_platforms_section.py [hunk 2]
```diff
@@ -140,6 +141,7 @@ def _model_platforms_docker_errors_section_data():
                 'exposed_ports': [bool()],
                 'published_ports': [int()],
                 'ulimits': [int()],
+                'user': str(),
```
Comment (decentral1se) [inline test/unit/model/v2/test_platforms_section.py]: Missing the test that checks that an incorect type is validated.

**Comment by rockandska** [comment]
> None of that is set in stone and no one has the strategy in place or has commited to the time needed to make such a wide sweeping change. Therefore, that remains a "hope". We need to stick to the coverage / testing approach that is in place now.

Off topic:
@decentral1se / @ssbarnea : I submit a proposal with a working example in #1923 for this kind of implementation but it seems no one has tried it or take the time to review it

Regards,

**Pull request merged**